pyro/poutine/block_messenger.py

```diff
@@ -77,47 +74,38 @@ class BlockMessenger(Messenger):
         3. ``msg["name"] not in expose and msg["type"] not in expose_types``
         4. ``hide``, ``hide_types``, and ``expose_types`` are all ``None``
 
-
     For example, suppose the stochastic function fn has two sample sites "a" and "b".
-    Then any poutine outside of BlockMessenger(fn, hide=["a"])
+    Then any effect outside of ``BlockMessenger(fn, hide=["a"])``
     will not be applied to site "a" and will only see site "b":
 
-    .. doctest::
-       :hide:
-
-       >>> from pyro.poutine.trace_messenger import TraceMessenger
-
-    >>> def fn():
-    ...     a = pyro.sample("a", dist.Normal(0., 1.))
-    ...     return pyro.sample("b", dist.Normal(a, 1.))
-
-    >>> fn_inner = TraceMessenger()(fn)
-    >>> fn_outer = TraceMessenger()(BlockMessenger(hide=["a"])(TraceMessenger()(fn)))
-    >>> trace_inner = fn_inner.get_trace()
-    >>> trace_outer  = fn_outer.get_trace()
-    >>> "a" in trace_inner
-    True
-    >>> "a" in trace_outer
-    False
-    >>> "b" in trace_inner
-    True
-    >>> "b" in trace_outer
-    True
-
-    See the constructor for details.
-
+        >>> def fn():
+        ...     a = pyro.sample("a", dist.Normal(0., 1.))
+        ...     return pyro.sample("b", dist.Normal(a, 1.))
+        >>> fn_inner = pyro.poutine.trace(fn)
+        >>> fn_outer = pyro.poutine.trace(pyro.poutine.block(fn_inner, hide=["a"]))
+        >>> trace_inner = fn_inner.get_trace()
+        >>> trace_outer  = fn_outer.get_trace()
+        >>> "a" in trace_inner
+        True
+        >>> "a" in trace_outer
+        False
+        >>> "b" in trace_inner
+        True
+        >>> "b" in trace_outer
+        True
+
+    :param fn: a stochastic function (callable containing Pyro primitive calls)
     :param hide_fn: function that takes a site and returns True to hide the site
         or False/None to expose it.  If specified, all other parameters are ignored.
         Only specify one of hide_fn or expose_fn, not both.
     :param expose_fn: function that takes a site and returns True to expose the site
         or False/None to hide it.  If specified, all other parameters are ignored.
         Only specify one of hide_fn or expose_fn, not both.
-    :param bool hide_all: hide all sites
-    :param bool expose_all: expose all sites normally
-    :param list hide: list of site names to hide, rest will be exposed normally
-    :param list expose: list of site names to expose, rest will be hidden
-    :param list hide_types: list of site types to hide, rest will be exposed normally
-    :param list expose_types: list of site types to expose normally, rest will be hidden
+    :param hide: list of site names to hide
```

Comment:
I think the previous docstring was more complete - `hide_all` and `expose_all` parameters are missing. Also maybe we should put in the type info back - some IDEs will do type checks based on the docstring.

Reply:
Feel free to address this in your followup PR, btw.

Reply:
Good catch, I'll include this in the followup PR